### DISCUSSION TITLE
Reproducibility of distributions in parameters variation (Randomness)

### DISCUSSION THREAD

**Discussion by rikkertKL** [the-anylogic-company/AnyLogic-Pypeline#70] [Q&A]
Dear, 

Seemingly the Anylogic - Python connection disturbs the reproducibility of distribution outcomes in my Anylogic model. I am generating Customer orders based on a distribution, set by the getDefaultRandomGenerator().setSeed(getCurrentReplication()); before each simulation run and called as double u = main.getDefaultRandomGenerator().nextDouble() ; within the distribution functions. Additionally, the randomness of the experiment itself is set as: new Random(getCurrentReplication()). After each iteration replications = 0; and after each simulation run replications++;. When running the experiment for multiple iterations with same replication count the number of the generated customer orders differs amongst replications with same value in different iterations. 

Now, when running the same experiment without interference of the pythonCommunicator the different simulations per iteration do provide the same values for used distributions. When adding the pythonCommunicator the generated values from distributions are slightly different, both from one another as from the same experiment without pythonCommunicator. The pythonCommunicator logic does not interfere the with generating values from the distributions, it only uses the generated values for analysis and communicates the analysis back iteratively for a production order logic. The customer order generation logic are standalone distribution functions called in a source block. 

Do you have previously encountered that Pypline may interfere with the randomness and so reproducibility of experiments without directly communicating with the model elements that generate randomness? Thank you in advance.

Best regards, 
Rik

**Comment by t-wolfeadam** [comment]
Hi! Sorry about the half-year delay, didn't get a notification for this apparently.

But I was unable to replicate what you've described. I tried creating both a 'Main' and 'MainPy' where the latter has a PyCommunicator agent which just adds up the total delay time in a simple queueing model; the regular 'Main' is the same but it does this purely via Java. I tested with two different param variations and they outputted the same values. Here is is: [Model396.zip](https://github.com/user-attachments/files/16057530/Model396.zip)

My guess is that there has to be some impact that Python is having on your model? That, or there's another hidden component to this.
